app/models/main.py

```diff
@@ -1847,8 +1847,8 @@ def validate(self, enforce_required=True, required_fields=None, max_day_rate=Non
         if errs:
             raise ValidationError(errs)
 
-    def serialize(self):
-        data = self.data.copy()
+    def serialize(self, with_data: bool = True):
+        data = {k: v for k, v in self.data.items() if with_data or k == "essentialRequirementsMet"}
```

Comment:
I think a comment here about when this field may or may not appear might be useful for future detectives, complementing the commit message saying why it's needed in the response. Something like `# 'essentialRequirementsMet' is present in brief responses from DOS2 onwards`?

Reply:
Yes - well, `with_data` is a bit of a dumb argument name anyway, but it seemed like the most systematic name and I couldn't think of anything better. (After all I wasn't the one that started this whole thing of calling this and that important variable/field `data`)

Reply:
I think it's fine for the arg itself, just the parametrised var name in the test was confusing me (about what we were expecting in each case)